stx-genesis/src/lib.rs

```diff
@@ -108,24 +108,32 @@ fn read_lockups(deflate_bytes: &'static [u8]) -> Box<dyn Iterator<Item = Genesis
 }
 
 fn read_namespaces(deflate_bytes: &'static [u8]) -> Box<dyn Iterator<Item = GenesisNamespace>> {
-    let namespaces = iter_deflated_csv(deflate_bytes).map(|cols| GenesisNamespace {
-        namespace_id: cols[0].to_string(),
-        importer: cols[1].to_string(),
-        buckets: cols[2].to_string(),
-        base: cols[3].parse::<i64>().unwrap(),
-        coeff: cols[4].parse::<i64>().unwrap(),
-        nonalpha_discount: cols[5].parse::<i64>().unwrap(),
-        no_vowel_discount: cols[6].parse::<i64>().unwrap(),
-        lifetime: cols[7].parse::<i64>().unwrap(),
+    let namespaces = iter_deflated_csv(deflate_bytes).map(|cols| {
+        let bucket_index = if cols.len() >= 10 { 4 } else { 2 };
```

Comment:
Is there an example where `cols.len() < 10`?

Reply:
The `chainstate.txt` file is that case, so it's tested by the `full-genesis` integration test.